src/nix/installables.cc

```diff
@@ -96,7 +96,15 @@ struct InstallableStorePath : Installable
 
     Buildables toBuildables() override
     {
-        return {{isDerivation(storePath) ? storePath : "", {{"out", storePath}}}};
+        if (isDerivation(storePath)) {
+            std::map<std::string, Path> outputs;
+            auto drv = readDerivation(storePath);
```

Comment:
I'm not sure this is the most optimal way to query the outputs, but it works for diverted/remote stores from what I can tell.